pennylane/templates/subroutines/hilbert_schmidt.py

```diff
@@ -72,103 +74,148 @@ class HilbertSchmidt(Operation):
-                return (1 - hilbert_test(v_params=parameters, v_function=v_function, v_wires=v_wires, u_tape=u_tape)[0])
+            def cost_hst(V, U):
+                return 1 - hilbert_test(V, U)[0]
 
-        Now that the cost function has been defined it can be called for specific parameters:
+        Now that the cost function has been defined it can be called as follows:
 
-        >>> cost_hst([0], v_function = v_function, v_wires = [1], u_tape = u_tape)
-        tensor(1., requires_grad=True)
+        >>> cost_hst(V, U)
+        np.float64(1.0)
 
     """
 
     grad_method = None
 
-    def _flatten(self):
-        metadata = (
-            ("v_function", self.hyperparameters["v_function"]),
-            ("v_wires", self.hyperparameters["v_wires"]),
-            ("u_tape", self.hyperparameters["u_tape"]),
-        )
-        return self.data, metadata
-
     @classmethod
-    def _primitive_bind_call(cls, *params, v_function, v_wires, u_tape, id=None):
+    def _primitive_bind_call(cls, V, U, **kwargs):  # kwarg is id
         # pylint: disable=arguments-differ
-        kwargs = {"v_function": v_function, "v_wires": v_wires, "u_tape": u_tape, "id": id}
-        return cls._primitive.bind(*params, **kwargs)
+        # AbstractOperator cannot be imported from the capture module
+        # because of a circular import, so we rely on this check instead.
+        U = (U,) if not hasattr(U, "__iter__") or is_abstract(U) else U
+        V = (V,) if not hasattr(V, "__iter__") or is_abstract(V) else V
+        num_v_ops = len(V)
+        return cls._primitive.bind(*V, *U, num_v_ops=num_v_ops, **kwargs)
 
-    @classmethod
-    def _unflatten(cls, data, metadata):
-        return cls(*data, **dict(metadata))
-
-    def __init__(self, *params, v_function, v_wires, u_tape, id=None):
-        self._num_params = len(params)
-
-        if not isinstance(u_tape, QuantumScript):
-            raise QuantumFunctionError("The argument u_tape must be a QuantumTape.")
-
-        u_wires = u_tape.wires
-        self.hyperparameters["u_tape"] = u_tape
-
-        if not callable(v_function):
-            raise QuantumFunctionError(
-                "The argument v_function must be a callable quantum function."
-            )
-
-        self.hyperparameters["v_function"] = v_function
+    def _flatten(self):
+        data = (self.hyperparameters["V"], self.hyperparameters["U"])
+        return data, tuple()
 
-        v_tape = make_qscript(v_function)(*params)
-        self.hyperparameters["v_tape"] = v_tape
-        self.hyperparameters["v_wires"] = Wires(v_wires)
+    @classmethod
+    def _unflatten(cls, data, _) -> "HilbertSchmidt":
+        return cls(*data)
+
+    def __init__(
+        self,
+        V: Operation | Iterable[Operation],
+        U: Operation | Iterable[Operation],
+        id: str | None = None,
+    ) -> None:
+
+        u_ops = (U,) if isinstance(U, Operation) else tuple(U)
```

Comment:
What if it's a `Prod`?
```suggestion
        u_ops = (U,) if isinstance(U, Operator) else tuple(U)
```

Reply:
Excellent catch, thanks (I still get confused with opmath ops...). I replaced `Operator` everywhere